src/app/(proper_react)/redesign/(authenticated)/user/welcome/View.tsx

```diff
@@ -51,7 +57,9 @@ export const View = ({
         onScanStarted={() => setCurrentStep("findExposures")}
         // TODO: Add unit test when changing this code:
         /* c8 ignore next */
-        onGoBack={() => setCurrentStep("getStarted")}
+        onGoBack={() =>
+          skipInitialStep ? router.back() : setCurrentStep("getStarted")
```

Comment:
Hmm... I'm a bit wary of relying on whatever the history state is - especially since there might not be a previous location. Don't we know the page the user should be coming from, and if so, could we `router.push` that?

Reply:
We do not know for sure where users are coming from. We are at least linking here from the dashboard and resolution flow. Alternatively, we could hide the back button if there is no previous history state and/or also add a global hook to track the previous location client-side?

Reply:
Yes, I was going to suggest that, but I didn't know of a way to check if there's no previous history state :P But if that's possible, that sounds like a good solution too.

(Though are you sure that we're linking straight to the EnterInfo page from the dashboard?)

Reply:
We could check the history length, which still might result in false positives because it includes forward items as well. A global hook might be more accurate?

I’m sure about linking from the dashboard directly to the `EnterInfo` page. Checked with Product and UX and added the link [here](https://github.com/mozilla/blurts-server/pull/3541/files#diff-fd7f08410ec922928b6ba022ee3ea2b999b1a0b02c96aea074710c28a65835f9R191).

Reply:
The main thing I'd be worried about w.r.t. the global hook is needing to remember using it everywhere, and in general it cluttering up the code of every page. It feels like something `useRouter` should be able to support it - might be worth looking in the Next.js issue tracker? Otherwise, the false positives feel like an edge case of an edge case, so is probably fine too.

Reply:
Already [checked](https://github.com/vercel/next.js/discussions/36723), but no luck. Ok, sounds good!

Reply:
So, it turns out everything I’ve tried first resulted in a potentially frustrating user experience. My preference right now would be to try and keep track of the previous route within Monitor. I’ve implemented a suggestion in PR #3621.

Reply:
Hmm, one downside of that approach is that React is going to re-render everything in that context provider (i.e. the entire app) on every path change.

One more attempt at an alternative approach: couldn't the previous page just pass a query parameter to indicate what page to link back to? e.g. `?referrer=dashboard`? And then if none is passed, we just leave out the button?

Reply:
Yeah, that works — I’ll be following up with this.

Reply:
Updated #3621 to use a referrer query param.